src/App.tsx

```diff
@@ -1,10 +1,49 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { USER_ID, getTodos } from './api/todos';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { TodoAppHeader } from './components/TodoAppHeader';
+import { TodoAppMain } from './components/TodoAppMain';
+import { TodoAppFooter } from './components/TodoAppFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Todo } from './types/Todo';
+import { ErrorText } from './types/ErrorText';
+import { StatusFilter } from './types/StatusFilter';
+import { wait } from './utils/fetchClient';
+import { getFilteredTodos } from './utils/getFilteredTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorText, setErrorText] = useState<ErrorText>(ErrorText.NoError);
+  const [statusFilter, setStatusFilter] = useState<StatusFilter>(
+    StatusFilter.All,
+  );
+
+  const preparedTodos = getFilteredTodos(todos, statusFilter);
+
+  const handleStatusFilterClick = (statusFilterValue: StatusFilter) => {
+    setStatusFilter(statusFilterValue);
+  };
+
+  const handleHideError = () => {
+    setErrorText(ErrorText.NoError);
+  };
+
+  const handleError = useCallback((message: ErrorText) => {
+    setErrorText(message);
+    wait(3000).then(() => handleHideError());
+  }, []);
+
+  useEffect(() => {
+    getTodos().then(
+      data => {
+        setTodos(data);
+      },
+      () => {
+        handleError(ErrorText.Loading);
+      },
+    );
```

Comment:
```suggestion
    getTodos()
      .then(setTodos)
      .catch(() => {
        handleError(ErrorText.Loading);
      });
```
Handle errors in the `catch` block